config/config.go

```diff
@@ -265,9 +265,9 @@ func (cfg *Config) normallizeTask(taskCfg *TaskConfig) (err error) {
 		taskCfg.Parser = "fastjson"
 	}
 
-	for _, dim := range taskCfg.Dims {
-		if dim.SourceName == "" {
-			dim.SourceName = util.GetSourceName(dim.Name)
+	for i := range taskCfg.Dims {
+		if taskCfg.Dims[i].SourceName == "" {
```

Comment:
`dim := taskCfg.Dims[i]`
Then use dim var

Reply:
1) you meant `dim := &taskCfg.Dims[i] `? 
2) we could, but I thought it's better to leave original code here

Reply:
1. Is Dims[i] has string type? Do you need pointer to string here?
2. You will resolve array index tree times

Reply:
as I wrote in PR description, dim is a struct

Reply:
array index is fast operation rather than pointer deref.

either way, this conversation is useless. config is parsed only once :)